### PR TITLE
Migrate core/workspace_audio.js to goog.module syntax

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test` locally already.

### Resolves

Part of #5026

### Proposed Changes

Converts `core/workspace_audio.js` to `goog.module` syntax.

### CODE REVIEW

File: core/workspace_audio.js [hunk 1]
```diff
@@ -98,10 +99,10 @@ Blockly.WorkspaceAudio.prototype.load = function(filenames, name) {
  * @package
  */
 Blockly.WorkspaceAudio.prototype.preload = function() {
-  for (var name in this.SOUNDS_) {
-    var sound = this.SOUNDS_[name];
+  for (let name in this.SOUNDS_) {
```
Comment (BeksOmega) [inline core/workspace_audio.js:101]: I think we decided to have these be `const`, or is that only for `for ... of` loops?

File: core/workspace_audio.js [hunk 2]
```diff
@@ -11,10 +11,10 @@
  */
 'use strict';
 
-goog.provide('Blockly.WorkspaceAudio');
+goog.module('Blockly.WorkspaceAudio');
+goog.module.declareLegacyNamespace();
 
 goog.require('Blockly.internalConstants');
-goog.require('Blockly.utils');
```
Comment (BeksOmega) [inline core/workspace_audio.js:17]: Just checking this was intentional.

**Review by BeksOmega — APPROVED** [review]
LGTM!

File: core/workspace_audio.js (hunk 2 above)
Comment (moniika) [reply core/workspace_audio.js:17]: Yes. It's not used in this file and, as far as I can tell, this extra module doesn't have any side effects to require importing.

File: core/workspace_audio.js (hunk 1 above)
Comment (moniika) [reply core/workspace_audio.js:101]: Updated.

**Pull request merged**